src/services/jsTyping.ts

```diff
@@ -25,7 +26,7 @@ namespace ts.JsTyping {
 
     // A map of loose file names to library names
     // that we are confident require typings
-    let safeList: Map<string>;
+    let safeList: { notFound: boolean, content?: Map<string> };

```

Comment:
You could use the distinction between undefined and null, (i.e. `safeList === undefined` means we haven't tried to load it, and `safeList === null` means we tried and there is no safe list).